pysam/libcalignmentfile.pyx

```diff
@@ -1538,14 +1538,15 @@ cdef class AlignmentFile(HTSFile):
         cdef int c = 0
         cdef int filter_method = 0
 
-        cdef int not_check_qual = 0
+        # check base quality if quality == None or 0
+        cdef int check_qual = 1 if quality_threshold else 0 
 
         if read_callback == "all":
             filter_method = 1
         elif read_callback == "nofilter":
             filter_method = 2
 
-        cdef int _threshold = quality_threshold
+        cdef int _threshold = quality_threshold if quality_threshold else 0
```

Comment:
should work with `quality_threshold or 0`

Reply:
thanks! does that mean setting ```quality_threshold = None``` should not be accepted and should raise an error?

Reply:
No.  The intent of `None` is to have no quality threshold (basically the same as zero).
```python
>>> None or 0
0
```

Reply:
so is the current line acceptable?  I think the if-clause is needed here to manually set ```None``` to 0, otherwise, ```cdef int _threshold = None``` will raise a TypeError.  And I also have a question on this line, I have checked ```_threshold``` is not used in any other place in the function, so I am guessing this line is just for making sure ```quality_threshold``` is set to an integer, right?

Reply:
Does this not work?  It is more succinct and should handle `None`s properly. 
```python
    cdef int _threshold = quality_threshold or 0
```

Reply:
it works! no idea there was this trick. thanks!